describe/describe.py

```diff
@@ -2,133 +2,244 @@
+MAX_SIZE_OF_DF = 5000
 
 
-def summarize(
+def analysis(
     context: MLClientCtx,
-    table: DataItem,
+    name: str = "dataset",
+    table: DataItem = None,
     label_column: str = None,
-    class_labels: List[str] = [],
-    plot_hist: bool = True,
     plots_dest: str = "plots",
-    update_dataset=False,
+    update_dataset: bool = False,
+    frac: float = 0.10,
 ) -> None:
-    """Summarize a table
-
-    :param context:         the function context
-    :param table:           MLRun input pointing to pandas dataframe (csv/parquet file path)
-    :param label_column:    ground truth column label
-    :param class_labels:    label for each class in tables and plots
-    :param plot_hist:       (True) set this to False for large tables
-    :param plots_dest:      destination folder of summary plots (relative to artifact_path)
-    :param update_dataset:  when the table is a registered dataset update the charts in-place
     """
+    The function will output the following artifacts per
+    column within the data frame (based on data types):
+
+    histogram matrix chart
+    histogram per feature chart
+    violin chart
+    correlation-matrix chart
+    correlation-matrix csv
+    imbalance pie chart
+    imbalance-weights-vec csv
+
+    :param context:                 the function context
+    :param name:                    key of dataset to database ("dataset" for default)
+    :param table:                   MLRun input pointing to pandas dataframe (csv/parquet file path)
+    :param label_column:            ground truth column label
+    :param plots_dest:              destination folder of summary plots (relative to artifact_path)
+                                    ("plots" for default)
+    :param update_dataset:          when the table is a registered dataset update the charts in-place
+    :param frac:                    when the table has more than 5000 samples,
+                                    the function will execute on random frac from the data (0.1 default)
+    """
+
     df = table.as_df()
-    header = df.columns.values
+    if df.shape[0] > MAX_SIZE_OF_DF:
+        df = df.sample(frac=frac)
     extra_data = {}
 
+    _create_histogram_mat_artifact(
+        context, df, extra_data, label_column, plots_dest
+    )
+    _create_features_histogram_artifacts(
+        context, df, extra_data, label_column, plots_dest
+    )
+    _create_violin_artifact(
+        context, df, extra_data, plots_dest
+                            )
+    _create_imbalance_artifact(
+        context, df, extra_data, label_column, plots_dest
+                               )
+    _create_corr_artifact(
+        context, df, extra_data, label_column, plots_dest
+                          )
+
+    artifact = table._artifact_url
+    if artifact is None:  # dataset not stored
+        artifact = context.log_dataset(key="dataset", db_key=name, stats=True, df=df)
+
+    if update_dataset:
```

Comment:
The idea was that this flag will be turned on if the dataset is logged, otherwise it is False. It shouldn't be an input from the user.

If dataset given is logged -> Update its extra data with the charts. 
Otherwise -> Log the dataset with the charts in extra data already (`log_dataset` should have `extra_data` argument available).